### ISSUE TITLE
Failed to purge service instance

### ISSUE THREAD

**Issue by bryancuster** [cloudfoundry/cli#690]
We have a corrupt/broken service that we can not delete. I have attempted to use the new `cf purge-service-instance` command, but I am receiving a 404.

```
name               service            plan             bound apps    last operation   
internal-redis   internal-redis   shared-vm   cf-redis           create succeeded   

$ cf purge-service-instance internal-redis
WARNING: This operation assumes that the service broker responsible for this service instance is no longer available or is not responding with a 200 or 410, and the service instance has been deleted, leaving orphan records in Cloud Foundry's database. All knowledge of the service instance will be removed from Cloud Foundry, including service bindings and service keys.

Really purge service instance internal-redis from Cloud Foundry?> yes

Purging service internal-redis...

FAILED
The service broker rejected the request to http://10.177.116.11/v2/service_instances/2edac993-5137-446a-8a07-94b44a015d7e/service_bindings/1fcfbd34-f6d4-4855-96c2-326e5512c46c. Status Code: 404 Not Found, Body: {}
```

Here is also the trace:
```
{
  "metadata": {
    "guid": "f4226cd8-8db9-4f01-bf1c-97f7db2925c1",
    "created_at": "2015-11-23T15:42:22Z",
    "url": "/v2/jobs/f4226cd8-8db9-4f01-bf1c-97f7db2925c1"
  },
  "entity": {
    "guid": "f4226cd8-8db9-4f01-bf1c-97f7db2925c1",
    "status": "failed",
    "error": "Use of entity>error is deprecated in favor of entity>error_details.",
    "error_details": {
      "code": 10001,
      "description": "The service broker rejected the request to http://10.177.116.11/v2/service_instances/2edac993-5137-446a-8a07-94b44a015d7e/service_bindings/1fcfbd34-f6d4-4855-96c2-326e5512c46c. Status Code: 404 Not Found, Body: {}\n",
      "error_code": "CF-ServiceBrokerRequestRejected",
      "http": {
        "uri": "http://10.177.116.11/v2/service_instances/2edac993-5137-446a-8a07-94b44a015d7e/service_bindings/1fcfbd34-f6d4-4855-96c2-326e5512c46c",
        "method": "DELETE",
        "status": 404
      }
    }
  }
}
```

For further background, this is me attempting to read the services service-keys:
```
REQUEST: [2015-11-23T10:45:14-05:00]
GET /v2/service_instances/2edac993-5137-446a-8a07-94b44a015d7e/service_keys HTTP/1.1
Host: api.pdc.sys.paas.lmig.com
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: application/json
User-Agent: go-cli 6.14.0+2654a47 / darwin



RESPONSE: [2015-11-23T10:45:15-05:00]
HTTP/1.1 404 Not Found
Content-Length: 86
Content-Type: application/json;charset=utf-8
Date: Mon, 23 Nov 2015 15:45:00 GMT
Server: nginx
X-Cascade: pass
X-Cf-Requestid: 548dfb70-f939-47f5-5420-13e271767bfa
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: 583e1801-073e-4820-6549-024c9a39f45a::05fe9b95-d254-4885-ad15-9bc47f1eddbb

{
  "code": 10000,
  "description": "Unknown request",
  "error_code": "CF-NotFound"
}
FAILED
Server error, status code: 404, error code: 10000, message: Unknown request
```

**Comment by utako** [comment]
Hi @bryancuster,

When the CLI hits the `purge_service_instances` endpoint, CC recursively deletes the service instances, associated service bindings and service keys. It looks like your service broker is unable to find the service binding with guid `1fcfbd34-f6d4-4855-96c2-326e5512c46c`. 

If you are able to look at cc logs, you can grep for `cc.service_lifecycle.instance_purger` to find the purging logs. You can also grep for `cc.background` plus the service binding guid to find logs on the service binding deletion delayed job, which may provide some more info on what's going on with the service broker.

Sincerely,
@swetharepakula && @utako
CF CAPI Team

**Comment by utako** [comment]
Hi again @bryancuster,

One more thought -- what version of CC API are you on? It is very likely that you are on an old version  that does not support this endpoint. We introduced this feature in CC API version 2.36.0. If you are indeed on an older version, I would try updating and trying again.

[More info here](https://github.com/cloudfoundry/cf-release/releases/tag/v218).

Thanks,
@swetharepakula && @utako
CF CAPI Team

**Comment by krishicks** [comment]
@utako @bryancuster This feature only working on 2.36.0 means there's a bug in the cli: it should fail gracefully when run against CC API <2.36.0.

**Comment by krishicks** [comment]
@bryancuster Do you have an update regarding your CC API version? I'd like to confirm that was your issue.

**Comment by bryancuster** [comment]
The version of the api is 2.23.0, so that makes sense! I'll have to try again once we update to release 225 or later.

**Comment by krishicks** [comment]
Excellent, thanks!

**Comment by krishicks** [comment]
@bryancuster: I'd like to close this bug. Could you download the latest edge release and let me know if that gives you a better error message?

**Comment by krishicks** [comment]
You can get the edge release from github.com/cloudfoundry/cli#downloads.

**Comment by dkoper** [comment]
This fix is included in cf CLI release v6.14.1 published today.
https://github.com/cloudfoundry/cli#downloads
